security/fma-contract-version-upgrade.md

```diff
@@ -0,0 +1,122 @@
+- **Recovery Path(s)**: Redeployment and upgrade of the affected contracts
+
+### Storage write removal before conditional termination (Bug fixed after 0.8.15)
+
+- **Description:**
+
+  - Introduced: v0.8.13
+  - Fixed: v0.8.17
+  - Explanation: Incorrect assumption that a storage write is redundant so the optimizer removes it. But this assumption can be wrong when the developer intended otherwise. A storage write is considered redundant when the storage slot is written to twice within an execution context without that slot being read before the second write or execution unconditionally terminates before the slot is read again.
+
+    According to the solidity blog, this bug can only occur if a contract contains pattern:
+
+    1. A storage write. Note that the write may still be removed even if it happens only conditionally or within a call to a function that ends up being inlined.
+    2. A call to a function that conditionally terminates using inline assembly as described above, but also has a different code path that returns to the caller.
+    3. Any continuing control flow path does one of the following:
+       - It overwrites the storage write in (1).
+       - It reverts.
+
+- **Risk Assessment:**
+  - Severity: High (if present)
+  - Likelihood: Low
+- **Mitigations:** Use a solidity version above 0.8.17.
+- **Detection:** If a solidity contract (using solidity version earlier than 0.8.17) follows the pattern described above, its potentially vulnerable to this.
+- **Recovery Path(s)**: Redeployment and upgrade of the affected contracts
+
+### Change in Default EVM version between different Solidity versions
+
+- **Description:** Version 0.8.25 sets the default evm version to cancun
+- **Risk Assessment:**
+  - Severity: Low/Medium/High
+  - Likelihood: Medium
+- **Mitigations:** As at 0.8.15, the default EVM version is 'London', if no explicit version override is declared, 'Cancun' will be used rather which if not intended can lead to code that either does not deploy or reverts un-ideally at runtime even though non-fork tests might have passed. A mitigation is explicitly declaring the intended EVM version to use for compilation regardless of if its the same as the default the compiler uses. That way, a new compiler version that changes the default EVM version does not change the one that is intended to be used. In this scenario however, this is not possible because the intended EVM version is the same as the compiler's EVM version default.
+- **Detection:** Running unit tests as fork test also.
+- **Recovery Path(s)**: Redeployment and upgrade of the affected contracts
+
+## Audit Requirements
+
+_Will this project require an audit according to the guidance in [OP Labs Audit Framework: When to get external security review and how to prepare for it](https://gov.optimism.io/t/op-labs-audit-framework-when-to-get-external-security-review-and-how-to-prepare-for-it/6864)? Please explain your reasoning._
```

Comment:
Let's fill this section out. My gut is that this does not need an audit, though offhand I'm not sure how it fits into the audit framework

Reply:
I see the thumbs up but looks like this rationale is still missing?